core/src/main/scala/org/http4s/Message.scala

```diff
@@ -269,14 +269,14 @@ final class Request[F[_]](
     )
 
   lazy val (scriptName, pathInfo) =
```

Comment:
What should we do about these?
We can make them into Uri.Paths and use the new splitAt(0), but then we need to change all routing to make sure it used the segment indices.
If we do that, then perhaps we need to change to a different backing collection type as random access might be more interesting